### ISSUE TITLE
Failed create rule constraint: Only admins can create new accounts with OAuth2

### ISSUE THREAD

**Issue by Buco7854** [henrygd/beszel#291]
Hello,
I setup OIDC with Authentik.
When I try to login with OIDC for the first time (did not try with already existing user)
I get the following log.
```
{
  "id": "rfkjzux469uhxje",
  "created": "2024-11-19 10:09:16.665Z",
  "updated": "2024-11-19 10:09:16.665Z",
  "data": {
    "auth": "guest",
    "details": "Failed create rule constraint: Only admins can create new accounts with OAuth2",
    "error": "Failed to authenticate.",
    "execTime": 709.867505,
    "method": "POST",
    "referer": "https://beszel.example.com/settings/general",
    "remoteIp": "192.168.1.61",
    "status": 400,
    "type": "request",
    "url": "/api/collections/users/auth-with-oauth2",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0",
    "userIp": "77.236.98.186"
  },
  "message": "POST /api/collections/users/auth-with-oauth2",
  "level": 8
}
```

**Comment by Buco7854** [comment]
Thanks a lot. Will there be a way to auto provision roles based on oidc claims?

**Comment by henrygd** [comment]
I'll look into it. For now every new registration should get the `user` role. If you need to set someone to `admin` or `readonly`, you can do so in the PocketBase users table.

Note that PocketBase admin accounts are separate from Beszel user accounts, so if you want someone to be able to log into PocketBase, you need to create an admin account for them at `/_/#/settings/admins`.

**Comment by Buco7854** [comment]
> I'll look into it. For now every new registration should get the `user` role. If you need to set someone to `admin` or `readonly`, you can do so in the PocketBase users table.
> 
> Note that PocketBase admin accounts are separate from Beszel user accounts, so if you want someone to be able to log into PocketBase, you need to create an admin account for them at `/_/#/settings/admins`.

Yes I did that, thanks, just wanted to know if it was planned. Pocketbase does not support oidc enrollment for admin sadly but it's fine.

Thanks for your time 😁.

**Comment by henrygd** [comment]
@MarkusGnigler I'm working on a docs site for the next release here: https://preview.beszel.dev

The OAuth functionality is native to PocketBase, so I don't know every detail on how it's implemented. But the accounts should be linked with a unique ID originating from the OAuth account (see screenshot). Are you having an issue with this?

![image](https://github.com/user-attachments/assets/c92e48f7-0ea1-4dbd-a925-681a72a26927)

I agree that user groups would be a nice time-saving feature if you have many users or systems. It's something I'd like to add in the future.

For now, perhaps using the REST API would be the most efficient way to assign users to systems. Here's an example using the PocketBase JS SDK with Bun:

```ts
import PocketBase from 'pocketbase'

const pb = new PocketBase('http://localhost:8090')

const systemNames = ['localhost', 'kagemusha']
const userEmails = ['user1@example.com', 'user2@example.com']

// authenticate as admin
await pb.admins.authWithPassword(process.env.EMAIL, process.env.PASSWORD)

// get user ids
const userIds = await pb
	.collection('users')
	.getFullList({
		fields: 'id',
		filter: `email='${userEmails.join(`'||email='`)}'`,
	})
	.then((records) => records.map(({ id }) => id))

// get id and current users for systems
const systemsData = await pb.collection('systems').getFullList({
	fields: 'id,users',
	filter: `name='${systemNames.join(`'||name='`)}'`,
})

// loop through systems and add users to them
for (const system of systemsData) {
	const updatedUsers = Array.from(new Set([...system.users, ...userIds]))
	await pb.collection('systems').update(system.id, { users: updatedUsers })
}
```

**Comment by MarkusGnigler** [comment]
Sorry for being late, had a complete server crash this weekend,

Thanks for pointing me to the documentation page.
Maybe I will find time to investigate the user mapping behavior of PocketBase. If I do, I'll report back.

Thanks for the snippet, and also for your friendly, open-minded personality for accepting all kinds of ideas!
